### PR TITLE
ncm-metaconfig: service snoopy

### PR DESCRIPTION
**Pull request by stdweird**
Add support for [snoopy](https://github.com/a2o/snoopy)

### CODE REVIEW

**Comment by jrha** [comment]
What is snoopy?

**Comment by stdweird** [comment]
@jrha i added project url in description

**Comment by jrha** [comment]
OMG why?

**Comment by stdweird** [comment]
to track users doing stupid things on shared interactive nodes

**Comment by jrha** [comment]
Fair enough :grin:

File: ncm-metaconfig/src/main/metaconfig/snoopy/pan/schema.pan [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+    match(SELF, '^(dev(log|null|tty)|syslog|std(out|err)|(file|socket):/.*)$')
+};
+
+type service_snoopy = {
+    'filter_chain' ? snoopy_filter_chain[]
+    'message_format' ? string
+    'output' ? snoopy_output
+    'error_logging' ? boolean
+    'syslog_facility' ? string with match(SELF, '^(LOG_)?(AUTH|AUTHPRIV|CRON|DAEMON|FTP|KERN|LOCAL[0-7]|LPR|MAIL|NEWS|SYSLOG|USER|UUCP)$')
+    'syslog_indent' ? string with (! match(SELF, '\s'))
```
Comment (jrha) [inline ncm-metaconfig/src/main/metaconfig/snoopy/pan/schema.pan]: What does this do? I can't find any reference to it elsewhere. Should it be `syslog_ident`?

File: ncm-metaconfig/src/main/metaconfig/snoopy/tests/profiles/config.pan [hunk 2]
```diff
@@ -0,0 +1,17 @@
+object template config;
+
+include 'metaconfig/snoopy/config';
+
+prefix "/software/components/metaconfig/services/{/etc/snoopy.ini}/contents";
+'message_format' = '[abc %{}  eswefwerwer]';
+'output' = 'socket:/some/path';
+'error_logging' = true;
+'syslog_facility' = 'LOG_CRON';
+'syslog_indent' = 'myname';
```
Comment (jrha) [inline ncm-metaconfig/src/main/metaconfig/snoopy/tests/profiles/config.pan]: Ditto.

File: ncm-metaconfig/src/main/metaconfig/snoopy/tests/regexps/config/base [hunk 3]
```diff
@@ -0,0 +1,13 @@
...
+---
+/etc/snoopy.ini
+---
+^\[snoopy\]$
+^error_logging = yes$
+^filter_chain = "only_root;exclude_uid:arg1,arg2"$
+^message_format = "\[abc %\{\}  eswefwerwer\]"$
+^output = "socket:/some/path"$
+^syslog_facility = "LOG_CRON"$
+^syslog_indent = "myname"$
```
Comment (jrha) [inline ncm-metaconfig/src/main/metaconfig/snoopy/tests/regexps/config/base]: Ditto.

**Comment by jrha** [comment]
@stdweird ping

**Comment by stdweird** [comment]
@jrha typo fixed

File: ncm-metaconfig/src/main/metaconfig/snoopy/tests/profiles/config.pan [hunk 4]
```diff
@@ -0,0 +1,17 @@
...
+'error_logging' = true;
+'syslog_facility' = 'LOG_CRON';
+'syslog_ident' = 'myname';
+'syslog_level' = 'EMERG';
+
+'filter_chain' = append(dict('filter', 'only_root'));
+'filter_chain' = append(dict(
+    'filter', 'exclude_uid',
+    'arguments', list('arg1', 'arg2'),
+    ));
```
Comment (jrha) [inline ncm-metaconfig/src/main/metaconfig/snoopy/tests/profiles/config.pan]: extra whitespace

File: ncm-metaconfig/src/test/perl/service-snoopy.t [hunk 5]
```diff
@@ -0,0 +1,13 @@
...
+BEGIN {
+    set_json_typed()
+}
+
+use Test::More;
+use Test::Quattor::TextRender::Metaconfig;
+
+my $u = Test::Quattor::TextRender::Metaconfig->new(
+        service => 'snoopy',
+        )->test();
```
Comment (jrha) [inline ncm-metaconfig/src/test/perl/service-snoopy.t]: four spaces please
too much indent on last line

**Comment by stdweird** [comment]
@jrha fixed

**Pull request merged**